src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -1576,11 +1576,10 @@ _public_ int sd_varlink_get_current_parameters(sd_varlink *v, sd_json_variant **
                 return -ENODATA;
 
         p = sd_json_variant_by_key(v->current, "parameters");
-        if (!p)
-                return -ENODATA;
 
+        /* For backward compatibility, return NULL when parameters field is missing (empty case) */
         if (ret)
-                *ret = sd_json_variant_ref(p);
+                *ret = p ? sd_json_variant_ref(p) : NULL;
```

Comment:
sd_json_variant_ref() happily accepts NULL and is a NOP in that case, returning NULL. thus the ternary op here is entirely redundant.